### PR TITLE
Muhammad Taufiq

### PR DESCRIPTION
**Pull request by Narwhalios**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?

3 hours

Please fill in one error and/or error message you received while working on this assignment.

'Something' is not defined

What part of the assignment did you spend the most time on?

To determine how the system know the rules of the game e.g scissors beat paper

Comfort Level (1-5):

3

Completeness Level (1-5):

3

What did you think of this deliverable?

It is okay, but there are things that can be improve on. For a start, the naming convention for the variables.

Is there anything in this code that you feel pleased about?

Getting the code to work after all the errors that I've went through

### CODE REVIEW

**Review by Narwhalios — COMMENTED** [review]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?

3 hours

Please fill in one error and/or error message you received while working on this assignment.

Getting the state from state 1 to state 2

What part of the assignment did you spend the most time on?

The 2 set of returns base on first message whether the user won that match and another where the system count the counters of wins and losing and draws.

Comfort Level (1-5):

3

Completeness Level (1-5):

4

What did you think of this deliverable?

It is better than I expected when I first start the assignment.

Is there anything in this code that you feel pleased about?

Getting the code to work after all the errors that I've went through

File: script.js [hunk 1]
```diff
@@ -1,4 +1,147 @@
+//Global for game states
+
+//2 states of the game
+var getUsername = "Username";
+var startSPSGame = "SPSgame";
+
+//Setting the conditions for game state one.
+var currentGameState = getUsername;
+var userName = "";
```
Comment (eddiejpot) [inline script.js:9]: Good use of gameStates!

File: script.js [hunk 2]
```diff
@@ -1,4 +1,147 @@
...
+  if (getNumberstoSigns == 1) {
+    return scissors;
+  }
+  if (getNumberstoSigns == 2) {
+    return paper;
+  }
+  if (getNumberstoSigns == 3) {
+    return stone;
+  }
+};
```
Comment (eddiejpot) [inline script.js:44]: Good work separating this function out!

File: script.js [hunk 3]
```diff
@@ -1,4 +1,147 @@
...
+var SPSGame = function (insert) {
+  var computerPick = getSigns();
+  var userPick = getUserPick(insert);
+  console.log("computer PICK");
+  console.log(computerPick);
+
+  var winMessage = `You won! You pick ${userPick} while Computer pick ${computerPick}.`;
+  var loseMessage = `You lose! You pick ${userPick} while Computer pick ${computerPick}.`;
+  var drawMessage = `Draw! Both pick ${computerPick}.`;
+  var invalidMessage = `Only type in scissors, paper or stone to play.`;
```
Comment (eddiejpot) [inline script.js:71]: Nice work using variables to store the sentences !

File: script.js [hunk 4]
```diff
@@ -1,4 +1,147 @@
...
+    return loseMessage;
+  }
+
+  if (userPick == computerPick) {
+    drawCounter = drawCounter + 1;
+    return drawMessage;
+  }
+  if (userPick != "scissors" || userPick != "paper" || userPick != "stone") {
+    return invalidMessage;
+  }
```
Comment (eddiejpot) [inline script.js:95]: Good use of conditionals and operators !

File: script.js [hunk 5]
```diff
@@ -1,4 +1,147 @@
...
+    ${drawCounter} draws. Nice!`;
+  }
+  if (userWinCounter < computerWinCounter) {
+    return ` You're losing the game! Computer's score is ${computerWinCounter}/${totalMatchesPlayed} while yours, ${userName} is ${userWinCounter}/${totalMatchesPlayed}. There's been ${drawCounter} draws. Go on and beat it!`;
+  }
+  if (userWinCounter == computerWinCounter) {
+    return ` It's a tie! The score is ${userWinCounter} / ${totalMatchesPlayed}. There's been ${drawCounter} draws.
+    Keep going ${userName}!`;
+  }
+};
```
Comment (eddiejpot) [inline script.js:113]: Wow nice work making a function to return the statements!

**Review by eddiejpot — COMMENTED** [review]
Overall great work! Looking forward to project 2 where we'll be using more data structures :) !

**Pull request closed without merging**